libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -1025,17 +1060,22 @@ const wxString& WaveClip::GetName() const
 
 sampleCount WaveClip::TimeToSamples(double time) const noexcept
 {
-    return sampleCount(floor(time * mRate + 0.5));
+   return sampleCount(floor(time * mRate / GetStretchRatio() + 0.5));
```

Comment:
I read this, and I want to change the function name to `DurationToSamples`.  If this change is correct, that better describes the meaning of the return value.  As with the distinction of durations from time points (as types!) in std::chrono.

Reply:
And then I would reexamine every use of the function to be sure they are correct.

Reply:
I often thought that having a type-safe differentiation between absolute and clip-relative times would spare us a few bugs and a little mental strain. I find the analogy between `std::chrono` time point and duration interesting, but I can't see how it fits here. What makes you say that `time` is a duration? `TimeToSamples` could very well be used to get the sample index closest to the time point `time`.

Reply:
> And then I would reexamine every use of the function to be sure they are correct.

I did.

Reply:
> And then I would reexamine every use of the function to be sure they are correct.

I did.